src/utility_thread.rs

```diff
@@ -0,0 +1,54 @@
+use std::time::Duration;
+
+use sqlx::PgPool;
+use tokio::time::sleep;
+use tokio::time::Instant;
+
+use crate::enterprise::directory_sync::get_directory_sync_interval;
+use crate::enterprise::{directory_sync::do_directory_sync, limits::do_count_update};
+
+const UTILITY_THREAD_MAIN_SLEEP_TIME: u64 = 5;
+const COUNT_UPDATE_INTERVAL: u64 = 60 * 60;
+
+pub async fn run_utility_thread(pool: &PgPool) -> Result<(), anyhow::Error> {
```

Comment:
1. start with `let mut last_... = Instanat::now()`
2. defined sleeping interval `let interval = Duration::from_secs(UTILITY_THREAD_MAIN_SLEEP_TIME);`
3. inside the loop, first thing to execute is sleep() (or tokio::time::interval if first loop shouldn't wait)
4. check elapsed() as before